graph-builder/src/registry.rs

```diff
@@ -249,6 +249,14 @@ pub fn fetch_releases(
         };
 
         if let Some(manifestref) = manifestref {
+            // Replace the tag specifier with the manifestref
+            release.source = {
+                let mut source_split: Vec<&str> = release.source.split(':').collect();
+                let _ = source_split.pop();
```

Comment:
digest-ref also [has a :](https://github.com/docker/distribution/blob/a25b9ef0c9fe242ac04bb20d3a028442b7d266b6/reference/reference.go#L17) within. I don't think there is any way for this `pop` to mistakenly operate on a hex-digest instead of a tag, but I wanted to mentioned that and make sure I didn't overlook something.

Reply:
this assumes that the current form is of `image:tag`, which I know it is at this point. Do you suggest to ensure that there's no `@` in the source?

Reply:
I think it's a fair assumption even without additional checks, we just need to remember and be careful in the future not to break it (as it is not captured by type-system checks).